### DISCUSSION TITLE
Client.Guid info ever sent back to Client?

### DISCUSSION THREAD

**Discussion by YGLim1961** [jchristn/WatsonTcp#274] [Q&A] [answered]
I like to ask if the Client.Guid (generated in ClientMetadata) sent back the client. Where can I capture that information on the client side. Thanks so much.

**Comment by jchristn** [comment]
Hi @YGLim1961 today it currently is not, but it could be in a pretty straight forward manner.  On the server side, you have the GUID available in the ```ConnectionEventArgs.Client``` object.  You could pass this in a message back to the client using either metadata or the message body.  Alternatively, another approach would be to send this back automatically (within the library).  If you have interest in integrating such a capability I'd love a PR on this!

**Comment by YGLim1961** [comment, marked as answer]
HI @jchristn 
Thanks for the reply. I think your suggestion is enough for my purpose. Again it is a wonderful library and I am slowly digging into the source code to understand it better.
Again my deep appreciation.
Regards
